src/interactions.rs

```diff
@@ -1,31 +1,53 @@
 use crate::github::{GithubClient, Issue};
 use std::fmt::Write;
 
-pub struct ErrorComment<'a> {
+pub struct Comment<'a> {
     issue: &'a Issue,
     message: String,
 }
 
+impl<'a> Comment<'a> {
+    pub fn new<T>(issue: &'a Issue, message: T) -> Comment<'a>
+    where
+        T: Into<String>,
+    {
+        Comment {
+            issue,
+            message: message.into(),
+        }
+    }
+
+    pub async fn post(&self, client: &GithubClient) -> anyhow::Result<()> {
+        self.issue.post_comment(client, &self.message).await
```

Comment:
I think we can just call this directly rather than have a struct for this.

Reply:
I was thinking that it might be better to have a general `Comment` struct that encapsulates commenting, and then `ErrorComment` can build on top of that.

Reply:
Would you mind elaborating on your reasoning a bit more?